### PR TITLE
[BUG #2189] Do not clear user set amount when scanning QR code

### PR DESCRIPTION
**Pull request by jeluard**
addresses #2189

### Summary:

QR code with no amount should not clear user set amount

### Steps to test:
- Open Status
- Navigate wallet/send
- Specify an amount
- Scan a QR code => original amount should be left intact in QR code doesn't contain amount

status: ready

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

File: src/status_im/ui/screens/wallet/choose_recipient/events.cljs [hunk 1]
```diff
@@ -11,7 +11,10 @@
       (assoc-in db [:wallet :send-transaction :camera-flashlight] toggled-state))))
 
 (defn choose-address-and-name [db address name amount]
-  (update-in db [:wallet :send-transaction] assoc :to-address address :to-name name :amount amount))
+  (cond-> db
+          true (update-in [:wallet :send-transaction] assoc :to-address address :to-name name)
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/choose_recipient/events.cljs]: just do `(cond-> (update-in db [:wallet :send-transaction] assoc :to-address address :to-name name)`

**Review by yenda — DISMISSED** [review]

File: src/status_im/ui/screens/wallet/choose_recipient/events.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/wallet/choose_recipient/events.cljs]: Good point!

**Comment by jeluard** [comment]
@yenda Updated

File: src/status_im/ui/screens/wallet/choose_recipient/events.cljs [hunk 2]
```diff
@@ -11,7 +11,9 @@
       (assoc-in db [:wallet :send-transaction :camera-flashlight] toggled-state))))
 
 (defn choose-address-and-name [db address name amount]
-  (update-in db [:wallet :send-transaction] assoc :to-address address :to-name name :amount amount))
+  (cond-> (update-in db [:wallet :send-transaction] assoc :to-address address :to-name name)
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/choose_recipient/events.cljs]:
```clojure
(update-in
   db [:wallet :send-transaction]
   #(cond-> (assoc % :to-address address :to-name name)
            amount (assoc :amount amount)))
```
probably a bit more compact

File: src/status_im/ui/screens/wallet/request/views.cljs [hunk 3]
```diff
@@ -32,7 +32,7 @@
 (views/defview qr-code [amount]
   (views/letsubs [account [:get-current-account]]
     [components.qr-code/qr-code
-     {:value   (eip67/generate-uri (:address account) {:value amount})
+     {:value   (eip67/generate-uri (:address account) (if amount {:value amount}))
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/request/views.cljs]: `when`

File: src/status_im/utils/eip/eip67.cljs [hunk 4]
```diff
@@ -32,9 +32,12 @@
...
+  "Generate a EIP 67 URI based on `address` and an optional map (keyword, string) of extra properties.
    No validation of address format is performed."
   ([address] (generate-uri address nil))
   ([address m]
    (when address
-     (str scheme scheme-separator address (when m (str parameters-separator (generate-parameter-string m)))))))
\ No newline at end of file
+     (let [parameters (into {} (filter second m))] ;; filter nil values
+       (str scheme scheme-separator address
+            (when-not (empty? parameters) (str parameters-separator (generate-parameter-string parameters))))))))
```
Comment (rasom) [inline src/status_im/utils/eip/eip67.cljs]: This line looks like it's longer than 80 symbols, can we move body to the next line?

File: src/status_im/ui/screens/wallet/choose_recipient/events.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/ui/screens/wallet/choose_recipient/events.cljs]: Looks better indeed.

**Comment by jeluard** [comment]
@rasom Made changes according to your comments

**Review by rasom — APPROVED** [review]

**Pull request merged**